Uploading CSV no longer works when there is materialized views (postgresql)

### Describe the bug

From version 0.51 (checked v0.51.4 as well, where there was a fixed for probably related issue https://github.com/metabase/metabase/issues/49912), if there is at least one materialized view, uploaded CSV model will have no fields or wrong fields detected.

### To Reproduce

1. Create a materialized view.
2. Upload any CSV
3. Open the model created from CSV
4. See error `no fields found for table <csv_table_name>`


### Expected behavior

_No response_

### Logs

<details>
<summary>ERROR: duplicate key value violates unique constraint "idx_unique_field"</summary>

```
metabase-1  | clojure.lang.ExceptionInfo: ERROR: duplicate key value violates unique constraint "idx_unique_field"
metabase-1  |   Detail: Key (name, table_id, unique_field_helper)=(created_at, 21, 0) already exists. {:toucan2/context-trace [["execute SQL with class com.mchange.v2.c3p0.impl.NewProxyConnection" {:toucan2.jdbc.query/sql-args ["INSERT INTO \"metabase_field\" (\"description\", \"database_type\", \"semantic_type\", \"table_id\", \"coercion_strategy\", \"name\", \"updated_at\", \"effective_type\", \"nfc_path\", \"parent_id\", \"database_partitioned\", \"database_is_auto_increment\", \"json_unfolding\", \"position\", \"visibility_type\", \"display_name\", \"database_position\", \"database_required\", \"created_at\", \"base_type\") VALUES ...
...
metabase-1  | 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2725)
...
metabase-1  | Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "idx_unique_field"
metabase-1  |   Detail: Key (name, table_id, unique_field_helper)=(created_at, 21, 0) already exists.
metabase-1  | 	... 428 more
```

</details>

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0",
    "vendor": ""
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-11-19",
      "tag": "v0.51.4",
      "hash": "47f05f3"
    },
    "settings": {
      "report-timezone": null
    },
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.7 (Debian 14.7-1.pgdg110+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.25+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.25",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.25+9",
    "os.name": "Linux",
    "os.version": "6.11.7-200.fc40.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  }
}
```

### Severity

Blocking upgrade from 0.50 to 0.51

### Additional context

_No response_

--- a/src/metabase/driver/postgres.clj
+++ b/src/metabase/driver/postgres.clj
@@ -339,8 +339,11 @@
       :join [[:pg_catalog.pg_namespace :pn] [:= :pn.oid :pc.relnamespace]
              [:pg_catalog.pg_attribute :pa] [:= :pa.attrelid :pc.oid]
              [:pg_catalog.pg_type :pt] [:= :pt.oid :pa.atttypid]]
-      :where [:and [:= :pc.relkind [:inline "m"]]
-              [:>= :pa.attnum 1]]}]
+      :where [:and
+              [:= :pc.relkind [:inline "m"]]
+              [:>= :pa.attnum 1]
+              (when schema-names [:in :pn.nspname schema-names])
+              (when table-names [:in :pc.relname table-names])]}]
     :order-by [:table-schema :table-name :database-position]}
    :dialect (sql.qp/quote-style driver)))
 

